ingestion/src/metadata/ingestion/source/dashboard/lightdash/client.py

```diff
@@ -0,0 +1,123 @@
+#  Copyright 2021 Collate
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+"""
+REST Auth & Client for Lightdash
+"""
+import traceback
+from typing import List
+
+from metadata.ingestion.ometa.client import REST, ClientConfig
+from metadata.ingestion.source.dashboard.lightdash.models import (
+    LightdashChart,
+    LightdashDashboard,
+)
+from metadata.utils.logger import utils_logger
+
+logger = utils_logger()
+
+
+class LightdashApiClient:
+    """
+    REST Auth & Client for Lightdash
+    """
+
+    client: REST
+
+    def __init__(self, config):
+        self.config = config
+        client_config = ClientConfig(
+            base_url=self.config.hostPort,
+            api_version="",
+            access_token=self.config.apiKey.get_secret_value(),
+            auth_header="Authorization",
+            auth_token_mode="ApiKey",
+            allow_redirects=True,
+        )
+        self.client = REST(client_config)
+
+    def get_org(self):
+        """GET api/org"""
+        return self.client.get(
+            "/api/v1/org",
+        )
+
+    def get_charts_list(self) -> List[LightdashChart]:
+        """
+        Get List of all charts
+        """
+        try:
+            response = self.client.get(
+                f"api/v1/projects/{self.config.projectUUID}/charts"
+            )
+            response_json_results = response["results"]
```

Comment:
We should do something like `results = response.get("results")` and then below check if `res` is None. Other we could end up with `Object of type "None" is not subscriptable` -- unless we have a guarantee on the API side to always have a `results` field

Reply:
done!